source/swap/contracts/Swap.sol

```diff
@@ -54,16 +56,21 @@ contract Swap is ISwap {
   // Mapping of signer addresses to an optionally set minimum valid nonce
   mapping (address => uint256) public signerMinimumNonce;
 
+  // contains registry of kind to asset types
+  TransferHandlerRegistry public registry;
+
   /**
     * @notice Contract Constructor
     * @dev Sets domain for signature validation (EIP-712)
+    * @param _registry TokenRegistry contract
     */
-  constructor() public {
+  constructor(TransferHandlerRegistry _registry) public {
```

Comment:
no underscore on parameters. Our style guide for constructors states
`Prefix all arguments with the name of the current contract.`
So it would be swapRegistry